#!/usr/bin/env python
PACKAGE = "mpc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#int
gen.add("predictedTargetNum_",    int_t,    0, "An Integer parameter", 8,  5, 100)
gen.add("mpc_steps_",    int_t,    0, "An Integer parameter", 50,  10, 200)
gen.add("mpc_w_cte_",    int_t,    0, "An Integer parameter", 500,  0, 9999)
gen.add("mpc_w_etheta_",    int_t,    0, "An Integer parameter", 500,  0, 9999)
gen.add("mpc_w_vel_",    int_t,    0, "An Integer parameter", 100,  0, 9999)
gen.add("mpc_w_angle_",    int_t,    0, "An Integer parameter", 100,  0, 9999)
gen.add("mpc_w_angle_d_",    int_t,    0, "An Integer parameter", 10000,  0, 9999)
gen.add("mpc_w_accel_",    int_t,    0, "An Integer parameter", 10,  0, 200)
gen.add("mpc_w_accel_d_",    int_t,    0, "An Integer parameter", 5,  0, 100)

#double
gen.add("passDistance_", double_t, 0, "A double parameter",    6.0, 0,   10.0)
gen.add("mpc_ref_vel_", double_t, 0, "A double parameter",    4.0, 0,   10.0)
gen.add("mpc_min_vel_", double_t, 0, "A double parameter",    4.0, 0,   10.0)



#bool
gen.add("islatencyCalculate_",   bool_t,   0, "A Boolean parameter",  False)
# gen.add("exit_bar_",   bool_t,   0, "A Boolean parameter",  False)
# gen.add("bool_save",   bool_t,   0, "A Boolean parameter",  False)

#string
# gen.add("save_name",    str_t,    0, "A string parameter",  "intersection_info")

# size_enum = gen.enum([ gen.const("Low",      int_t, 0, "Low is 0"),
#                        gen.const("Medium",     int_t, 1, "Medium is 1"),
#                        gen.const("High",      int_t, 2, "Hight is 2")],
#                      "Select from the list")

# gen.add("size", int_t, 0, "Select from the list", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "mpc", "mpc_"))
